applying endpoint configuration in quick succession fails

Due to changes made in https://github.com/cilium/cilium/pull/3043 , which fixed some issues found in https://github.com/cilium/cilium/pull/3040, applying endpoint configuration updates in quick succession caused some build failures:


Test failure on Jenkins:
```
23:33:30 Summarizing 1 Failure:
23:33:30 
23:33:30 [Fail] RuntimeValidatedConntrackTable [It] Testing L3-L4-L7 policy while the L3-L4 connection was previously made to confirm L3-L4-L7 policy enforcement over the proxy with the same source port 
23:33:30 /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/runtime/ct.go:627
23:33:30 
23:33:30 Ran 55 of 267 Specs in 3055.116 seconds
23:33:30 FAIL! -- 54 Passed | 1 Failed | 0 Pending | 212 Skipped --- FAIL: TestTest (3055.13s)
```

More specifically:
```
22:51:51 • Failure [41.844 seconds]
22:51:51 RuntimeValidatedConntrackTable
22:51:51 /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/runtime/ct.go:53
22:51:51   Testing L3-L4-L7 policy while the L3-L4 connection was previously made to confirm L3-L4-L7 policy enforcement over the proxy with the same source port [It]
22:51:51   /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/runtime/ct.go:617
22:51:51 
22:51:51   Setting 13233 endpoint's EgressPolicy as false
22:51:51   Expected
22:51:51       <bool>: false
22:51:51   to be true
22:51:51 
22:51:51   /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/runtime/ct.go:627
```

From the logs:
```
level=debug msg="running command: sudo cilium endpoint list -o jsonpath='{range [?(@.labels.orchestration-identity[0]!=\"reserved:health\")]}{@.container-name}{\"=\"}{@.id}{\"\\n\"}{end}'"
cmd: "sudo cilium endpoint list -o jsonpath='{range [?(@.labels.orchestration-identity[0]!=\"reserved:health\")]}{@.container-name}{\"=\"}{@.id}{\"\\n\"}{end}'" exitCode: 0 
 server-2=13233
server-3=13628
client-2=17471
netcat=29140
server=38593
client=50640


level=debug msg="running command: sudo cilium endpoint config 13233 IngressPolicy=false"
cmd: "sudo cilium endpoint config 13233 IngressPolicy=false" exitCode: 0 
 Endpoint 13233 configuration updated successfully

level=debug msg="running command: sudo cilium endpoint config 13233 EgressPolicy=false"
cmd: "sudo cilium endpoint config 13233 EgressPolicy=false" exitCode: 1 
 Error: Cannot update endpoint 13233: [PATCH /endpoint/{id}/config][500] patchEndpointIdConfigFailed  unable to regenerate endpoint program because state transition to waiting-to-regenerate was unsuccessful; check `cilium endpoint log 13233` for more information
```

This is directly due to a change caused by https://github.com/cilium/cilium/pull/3043 , in which we check the result of state changes upon configuration updates.

--- a/pkg/endpoint/endpoint.go
+++ b/pkg/endpoint/endpoint.go
@@ -1209,7 +1209,9 @@ func (e *Endpoint) Update(owner Owner, opts models.ConfigurationMap) error {
 			return nil
 		}
 
-		return fmt.Errorf("unable to regenerate endpoint program because state transition to %s was unsuccessful; check `cilium endpoint log %d` for more information", StateWaitingToRegenerate, e.ID)
+		// FIXME: GH-3058: We need to queue up a regeneration
+		// nevertheless
+		return nil
 	}
 
 	e.Mutex.Unlock()

